lib/handlers_v5.js

```diff
@@ -341,3 +344,21 @@ function shutdown_request(request) {
 
     status_idle.call(this, request);
 }
+
+function is_complete_request(request) {
+    try {
+        esprima.parse(request.content.code);
+    } catch(e) {
+        if (e.message.indexOf('Unexpected end of input') > -1) {
+            request.respond(this.shellSocket, "is_complete_reply", {
+                'status': 'incomplete',
+                'indent': '    '
```

Comment:
I meant to comment on the indentation last night, but I forgot. I think `indent` should be set to an empty string (""), because in Javascript indentation isn't semantic.